joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/parser/GoAstJsonParser.scala

```diff
@@ -17,4 +23,15 @@ object GoAstJsonParser {
     val sourceFileContent = IOUtils.readEntireFile(filePath)
     ParserResult(filePath.getFileName.toString, fullFilePath, json, sourceFileContent)
   }
+
+  def readModFile(file: Path): Option[GoMod] = {
+    val jsonContent = IOUtils.readLinesInFile(file).mkString
+    val t = decode[GoMod](jsonContent) match {
```

Comment:
The `t` variable is unnecessary here
```scala
    decode[GoMod](jsonContent) match {
       case Left(error) =>
         logger.warn(s"Error decoding JSON - '${file.toString}': $error")
         Left(error)
       case x => x
     }.toOption
```

Reply:
For some reason, `.toOption` is throwing a compilation error.

Reply:
Does the following perhaps work
```scala
    (decode[GoMod](jsonContent) match {
       case Left(error) =>
         logger.warn(s"Error decoding JSON - '${file.toString}': $error")
         Left(error)
       case x => x
     }).toOption
 ```

Reply:
done